### ISSUE TITLE
MacOS delays while playing

### ISSUE THREAD

**Issue by iwheelbuy** [ioquake/ioq3#346]
Hello.

I installed  and launched the game successfully. But it is impossible to play because of the delays / lags. The game freezes every second for half a second. And not only while actually playing. It behaves the same way while I’m moving the mouse on the main q3 menu.

I’m on High Sierra with quad core Mac mini late 2012. Is it my hardware, OS or I’m missing something?

Thanks

**Comment by Calinou** [comment]
Does the issue persist if you switch to the OpenGL 1.x renderer by entering `/cl_renderer opengl1; vid_restart` in the console?

**Comment by ensiform** [comment]
Test build of ioquake3?

**Comment by zturtleman** [comment]
I'm not sure if this is the same issue but a [post on the ioq3 forum](https://discourse.ioquake.org/t/macos-high-sierra-mouse-and-video-lag/1034) suggests rebooting to fix slow OpenGL on macOS High Sierra.

**Comment by iwheelbuy** [comment]
I will test all the suggested solutions and post all the results. Thanks

**Comment by iwheelbuy** [comment]
I haven't found where to get a **test build**. So I've built the ioquake3 from latest **master** commit and it doesn't help. Rebooting doesn't help. Switching to the OpenGL 1.x renderer doesn't help. I give up... Will try again when High Sierra 10.13.4 is released.

**Comment by TimeDoctor** [comment]
https://ioquake3.org/get-it/test-builds/

**Comment by grigorievk** [comment]
I have same issue, test builds doesn't help too.

**Comment by TimeDoctor** [comment]
@grigorievk do you have the same/similar hardware to the OP?

**Comment by iwheelbuy** [comment]
updated to 10.13.4. nothing changed

**Comment by grigorievk** [comment]
@TimeDoctor yes, Mac mini (Late 2012) (2,3 GHz Intel Core i7, 16 GB 1333 MHz DDR3, Intel HD Graphics 4000 1536 MB).

**Comment by ensiform** [comment]
Try with com_busyWait 1 and restart the game

**Comment by iwheelbuy** [comment]
I have tried with MacBook Pro 13'' 2014, Mac Mini 2014 and some old iMac. On each device the game goes perfectly. I think there is some issue with Mac Mini Late 2012

@ensiform will try, thanks

**Comment by kiliw** [comment]
I had a lot of lags too with my MacBook Pro (13", Mid 2012). After setting the OpenGL 1.x render everything was fine.
Here is my custom settings `autoexec.cfg file from the basq3 folder:
```
seta r_mode "-1"
seta r_customwidth "1280"
seta r_customheight "800"
seta cg_fov "115"
seta r_displayrefresh "0"
seta cl_renderer "opengl1"
```

**Comment by gubby** [comment]
I am able to repro this on my Mac mini (Late 2012) - and kiliw's settings above make no difference for me. I have tried all manner of settings and nothing seems to make a difference.

For me the symptom is a very short hiccup in the video 1-2 seconds or so. It feels like only a handful of frames are dropped. The FPS counter (cg_drawFPS) sees this as a drop to ~30fps for a moment, and then it recovers. The mouse also 'feels' laggy at the menu screen, but I wouldn't describe the motion as jerky.

The exact same binary and install dir (built manually from f2c61c14) have no problems on my MacBook (Retina, 12-inch 2017).

**Comment by ensiform** [comment]
@gubby pretty much every issue on Mac with games now is because of Apple making breaking changes of some kind in every new release due to not caring about any legacy software as a core principle. Pretty soon OpenGL support all together will almost certainly completely go the wayside too as it has already been deprecated for a year.

One suggestion is to build and link against the very latest SDL and revert to using opengl1 renderer maybe as well.

**Comment by tomkidd** [comment]
Due to the age of this issue I'm going to close it, let me know if it's still an issue and we can reopen it.
